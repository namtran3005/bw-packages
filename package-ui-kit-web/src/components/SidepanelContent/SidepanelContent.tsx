import * as React from 'react';
import styled from 'styled-components';

import { Content, ContentProps } from '../..';
import { RootContent as ContentContent } from '../Content/Content.styled';

interface SidepanelContentProps extends ContentProps {
  docked?: 'left' | 'right';
  justify?: React.CSSProperties['justifyContent'];
  compensateLogoHeight?: boolean;
}

const SidepanelComponentWrapper: React.FunctionComponent<SidepanelContentProps> = ({
  /* eslint-disable @typescript-eslint/no-unused-vars */
  docked,
  justify,
  compensateLogoHeight,
  /* eslint-enable @typescript-eslint/no-unused-vars */
  ...cleanProps
}) => <Content {...cleanProps} />;

const SidepanelComponent = styled(SidepanelComponentWrapper)`
  position: relative;
  flex: 1;
  display: flex;
  justify-content: center;
  ${({ docked }) => (docked ? `${docked}: -50%;` : '')};

  ${ContentContent} {
    width: 100%;
    height: 100%;
    margin-top: initial;
    margin-bottom: initial;
    display: flex;
    flex-direction: column;
    justify-content: ${({ justify }) => justify || 'start'};
  }
`;

const SidepanelContent: React.FunctionComponent<SidepanelContentProps> = (
  props
) => <SidepanelComponent {...props} />;

export { SidepanelContent, SidepanelContentProps };
