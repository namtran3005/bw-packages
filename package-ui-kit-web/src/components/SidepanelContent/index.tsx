export { SidepanelContent, SidepanelContentProps } from './SidepanelContent';
