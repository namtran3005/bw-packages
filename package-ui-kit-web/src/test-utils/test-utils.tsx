import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
