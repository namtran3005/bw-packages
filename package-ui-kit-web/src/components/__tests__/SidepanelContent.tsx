import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { SidepanelContent } from '../SidepanelContent';

describe('SidepanelContent', () => {
  it('should render SidepanelContent', () => {
    const { baseElement } = renderWithTestHarness(<SidepanelContent />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should render SidepanelContent docked left', () => {
    const { baseElement } = renderWithTestHarness(
      <SidepanelContent docked="left" />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render SidepanelContent docked right', () => {
    const { baseElement } = renderWithTestHarness(
      <SidepanelContent docked="right" />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
