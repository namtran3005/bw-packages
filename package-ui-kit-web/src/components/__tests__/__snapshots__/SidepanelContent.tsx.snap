// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidepanelContent should render SidepanelContent 1`] = `
.c0 {
  position: relative;
  margin-left: 0px;
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  height: auto;
  min-height: auto;
  z-index: initial;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
}

.c0 > .{
  box-sizing: border-box;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: inherit;
  margin-right: inherit;
  max-width: auto;
  text-align: inherit;
  height: auto;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .{
  width: 100%;
  height: 100%;
  margin-top: initial;
  margin-bottom: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

<body>
  <div>
    <div
      class="c0 c1"
    >
      <div
        class=""
      />
    </div>
  </div>
</body>
`;

exports[`SidepanelContent should render SidepanelContent docked left 1`] = `
<body>
  .c0 {
  position: relative;
  margin-left: 0px;
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  height: auto;
  min-height: auto;
  z-index: initial;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
}

.c0 > .{
  box-sizing: border-box;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: inherit;
  margin-right: inherit;
  max-width: auto;
  text-align: inherit;
  height: auto;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: -50%;
}

.c1 .{
  width: 100%;
  height: 100%;
  margin-top: initial;
  margin-bottom: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

<div>
    <div
      class="c0 c1"
    >
      <div
        class=""
      />
    </div>
  </div>
</body>
`;

exports[`SidepanelContent should render SidepanelContent docked right 1`] = `
<body>
  .c0 {
  position: relative;
  margin-left: 0px;
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  height: auto;
  min-height: auto;
  z-index: initial;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
}

.c0 > .{
  box-sizing: border-box;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: inherit;
  margin-right: inherit;
  max-width: auto;
  text-align: inherit;
  height: auto;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  right: -50%;
}

.c1 .{
  width: 100%;
  height: 100%;
  margin-top: initial;
  margin-bottom: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

<div>
    <div
      class="c0 c1"
    >
      <div
        class=""
      />
    </div>
  </div>
</body>
`;
